cpp/collections_module/algorithms/algorithms.cpp

```diff
@@ -0,0 +1,41 @@
+#include "algorithms.hpp"
+
+void Collections::RemoveAll(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try {
+    const auto input_list = arguments[0].ValueList();
+    const auto to_remove_list = arguments[1].ValueList();
+    std::set<mgp::Value> searchable;
+
+    for (const auto value : input_list) {
+      searchable.insert(std::move(value));
+    }
+
+    std::set<mgp::Value>::iterator itr;
+
+    for (const auto key : to_remove_list) {
+      while (true) {
+        itr = searchable.find(key);
+        if (itr == searchable.end()) {
+          break;
+        }
+        searchable.erase(itr);
+      }
+    }
```

Comment:
I am wondering now will this implementation work in case you have:

[1,2,3,2,4], [4]

because with the set, you lose cardinality. You can use std::multiset in that way so you are safe with cardinality of numbers